### PR TITLE
Horus simulator rotary control improvements 4085

### PR DESCRIPTION
**Pull request by joaquins**
A couple of proposal fixes:
- Add keys for rotary control (comma, period)
- Fix for wheelEvent function

### CODE REVIEW

File: companion/src/simulation/simulatordialog.cpp [hunk 1]
```diff
@@ -70,7 +70,10 @@ void SimulatorDialog::updateDebugOutput()
 
 void SimulatorDialog::wheelEvent (QWheelEvent *event)
 {
-  simulator->wheelEvent(event->delta() > 0 ? 1 : -1);
+   if ( event->delta() != 0)
```
Comment (3djc) [inline companion/src/simulation/simulatordialog.cpp]: cosmetics : 
if (...) {
  ...
}

**Comment by 3djc** [comment]
comma and period may not be a great choice for some language (one is lowercase, one is uppercase on a french keyboard for example)

**Comment by schwabe** [comment]
simu uses x and z (which are also not ideal)

**Comment by 3djc** [comment]
schwabe makes a good point, we should have consistency between both (knowing no key combination will be perfect for all keyboard layouts)

**Comment by joaquins** [comment]
Thanks for the comments guys. I already fixed the cosmetic issues, and changed the kay assignments to keys Z and X as suggested.

Is there any other step needed to get this pull request accepted, and mark the opened issue as fixed?

**Comment by 3djc** [comment]
I requires one of us to test it and approved it (or not). But we are having a very busy day, it may take some time

**Comment by joaquins** [comment]
Oh ok @3djc, thanks for your answer. No pressure at all, just wondering if I needed to do anything else.

File: companion/src/simulation/simulatordialog.cpp [hunk 2]
```diff
@@ -70,7 +70,9 @@ void SimulatorDialog::updateDebugOutput()
 
 void SimulatorDialog::wheelEvent (QWheelEvent *event)
 {
-  simulator->wheelEvent(event->delta() > 0 ? 1 : -1);
+   if ( event->delta() != 0) {
+       simulator->wheelEvent(event->delta() > 0 ? 1 : -1);
```
Comment (bsongis) [inline companion/src/simulation/simulatordialog.cpp]: wrong indentation

File: companion/src/simulation/simulatordialog.cpp [hunk 3]
```diff
@@ -220,6 +222,13 @@ void SimulatorDialog::keyPressEvent (QKeyEvent *event)
     case Qt::Key_PageUp:    
       buttonPressed = event->key();
       break;
+    case Qt::Key_Z:
```
Comment (bsongis) [inline companion/src/simulation/simulatordialog.cpp]: On FR keyboards this is not good, perhaps is it possible you check the keyboard layout?

**Comment by rcbebo82** [comment]
Works nice. This is really good it was a pain with the mouse wheel!

**Comment by schwabe** [comment]
If we move them to x and c (which also helps German Qwertz), could you do the same for simu.cpp to have the two programs consistent?

**Comment by joaquins** [comment]
Ping?
I already did the update to simu to have the same key mapping. Let me know if you are expecting something else.

**Comment by projectkk2glider** [comment]
I will test this now...

**Comment by projectkk2glider** [comment]
There are two issues with Horus in Companion:
 * mouse scroll-wheel works in reverse
 * PgUp works, but PgDown does not, Backspace works as PgDown.

@joaquins would you be willing to solve this two issues also as a part of this PR? Otherwise your chages work good.

**Comment by joaquins** [comment]
@projectkk2glider I would need to get a "real" mouse and a full keyboard for those... using laptop trackpad so far :(

**Comment by projectkk2glider** [comment]
OK, then I will merge this and we will deal with #4160 later

**Comment by projectkk2glider** [comment]
Thanks!

**Comment by joaquins** [comment]
Thanks @projectkk2glider !

**Pull request merged**